### PR TITLE
Added availability set for workers and security group to allow ingress access

### PR DESCRIPTION
**Pull request by dharapvj**
**What this PR does / why we need it**:
Currently, Azure terraform example has two issues:
1. If we install any service of type LoadBalancer, it does not provide LoadBalancer. this is because of Azure having some restriction on number of LoadBalancers in a given availability set. We are having a single AV set in the code for masters as well as workers and we already provision one load balancer for api-server. So we cannot provision any service of type LoadBalancer (e.g. nginx ingress)
2. If we get 2nd AVset for above issue and then install nginx-ingress, we cannot reach the ingress endpoints due to security group restrictions.

This PR solves both above issues by adding an additional avset for worker machines and also relaxing security group to reach on port 80 and 443 from LB.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

**Special notes for your reviewer**:
Partially related #1307 . Need documentation fix though.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by xmudrii — CHANGES_REQUESTED** [review]
Can you please edit the PR description to replace `Fixes #...` with `xref #...`? I don't want to close the issue yet because I think this should also be documented. I'll take care of the docs later.

File: examples/terraform/azure/output.tf [hunk 1]
```diff
@@ -59,7 +59,7 @@ output "kubeone_workers" {
           # see example under `cloudProviderSpec` section at: 
           # https://github.com/kubermatic/machine-controller/blob/master/examples/azure-machinedeployment.yaml
           assignPublicIP    = true
-          availabilitySet   = azurerm_availability_set.avset.name
+          availabilitySet   = azurerm_availability_set.avset_worker.name
```
[the remark this answers is not in the document]
Comment (dharapvj) [reply examples/terraform/azure/output.tf]: Done.

File: examples/terraform/azure/main.tf [hunk 2]
```diff
@@ -94,6 +108,28 @@ resource "azurerm_network_security_group" "sg" {
...
+    name                       = "Azure_LB_to_VMs_http"
+    priority                   = 1003
+    direction                  = "Inbound"
+    access                     = "Allow"
+    protocol                   = "Tcp"
+    source_port_range          = "*"
+    destination_port_range     = "80"
+    source_address_prefix      = "AzureLoadBalancer"
+    destination_address_prefix = "*"
+  }
```
[the remark this answers is not in the document]
Comment (dharapvj) [reply examples/terraform/azure/main.tf]: Done. Removed.

File: examples/terraform/azure/main.tf [hunk 3]
```diff
@@ -45,6 +45,20 @@ resource "azurerm_availability_set" "avset" {
   }
 }
 
+resource "azurerm_availability_set" "avset_worker" {
+  name                         = "${var.cluster_name}-avset-worker"
```
[the remark this answers is not in the document]
Comment (dharapvj) [reply examples/terraform/azure/main.tf]: Done.

File: examples/terraform/azure/main.tf [hunk 4]
```diff
@@ -94,7 +108,7 @@ resource "azurerm_network_security_group" "sg" {
     source_address_prefix      = "*"
     destination_address_prefix = "*"
   }
-
+
```
Comment (xmudrii) [inline examples/terraform/azure/main.tf:111]: Tiny nit: can you remove this space?

Comment (dharapvj) [reply examples/terraform/azure/main.tf]: Done.

**Review by xmudrii — APPROVED** [review]
Thanks, @dharapvj!

**Pull request merged**